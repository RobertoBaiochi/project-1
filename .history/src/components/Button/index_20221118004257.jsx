class Button extends Component {
  state = {  }
   
  render() { 
    return ();
  }
}
 
export default Button;