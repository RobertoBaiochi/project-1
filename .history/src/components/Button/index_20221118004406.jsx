class Button extends Component {
  state = {  }

  render() { 
    return (
      <button>
        Load More Post
      </button>
    );
  }
}
 
export  Button;