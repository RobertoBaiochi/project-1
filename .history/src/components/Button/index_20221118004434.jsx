
export class Button extends Component {
  state = {  }

  render() { 
    return (
      <button>
        Load More Post
      </button>
    );
  }
}
 