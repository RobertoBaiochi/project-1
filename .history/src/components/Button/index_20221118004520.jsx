import { Component } from 'react'

class Button extends Component {
  render() { 
    return (
      <button>
        Load More Post
      </button>
    );
  }
}
 
export default Button;