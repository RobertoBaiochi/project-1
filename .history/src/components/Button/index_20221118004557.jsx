import { Component } from 'react'

class Button extends Component {
  render() { 
    return (
      <button>
        {text}
      </button>
    );
  }
}
 
export default Button;