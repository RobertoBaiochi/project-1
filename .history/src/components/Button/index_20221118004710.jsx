import { Component } from 'react'

class Button extends Component {

  render() {
    
    const { text }

    return (
      <button>
        {text}
      </button>
    );
  }
}
 
export default Button;