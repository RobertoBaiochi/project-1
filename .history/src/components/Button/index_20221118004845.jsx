import { Component } from 'react'

class Button extends Component {

  render() {
    
    const { text, onClick } = this.props;

    return (
      <button onClick={}>
        {text}
      </button>
    );
  }
}
 
export default Button;