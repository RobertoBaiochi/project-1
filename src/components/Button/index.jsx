import P from 'prop-types';

import { Component } from 'react';

import './styles.css';

class Button extends Component {
  render() {
    const { text, onClick, disabled = false } = this.props;

    return (
      <button disabled={disabled} className="btn-pages" onClick={onClick}>
        {text}
      </button>
    );
  }
}

export default Button;

Button.propTypes = {
  text: P.string.isRequired,
  onClick: P.func.isRequired,
  disabled: P.bool,
};
