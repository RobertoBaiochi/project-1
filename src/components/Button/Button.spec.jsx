import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '.';

import TestRenderer from 'react-test-renderer';

describe('<Button />', () => {
  it('should render the button with the text "Load More"', () => {
    const fn = jest.fn();
    render(<Button text="Load More" onClick={fn} />);
    expect.assertions(2);

    const button = screen.getByRole('button', { name: /load more/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('class', 'btn-pages');
  });

  it('should call fuction on button click', () => {
    const fn = jest.fn();
    render(<Button text="Load More" onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });
    fireEvent.click(button);
    userEvent.click(button);
    expect(fn).toHaveBeenCalledTimes(2);
  });

  it('should be disabled when disabled is true', () => {
    const fn = jest.fn();
    render(<Button text="Load More" disabled={true} onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });

    expect(button).toBeDisabled();
  });

  it('should be enabled when disabled is false', () => {
    const fn = jest.fn();
    render(<Button text="Load More" disabled={false} onClick={fn} />);
    const button = screen.getByRole('button', { name: /load more/i });
    expect(button).toBeEnabled();
  });

  it('should match snapshot', () => {
    const fn = jest.fn();

    const testRenderer = TestRenderer.create(<Button text="Load More" disabled={false} onClick={fn} />).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });
});
