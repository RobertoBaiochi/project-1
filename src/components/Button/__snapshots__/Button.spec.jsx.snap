// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<button
  className="btn-pages"
  disabled={false}
  onClick={[MockFunction]}
>
  Load More
</button>
`;
